x-pack/plugins/security/server/authorization/disable_ui_capabilities.ts

```diff
@@ -67,20 +68,56 @@ export function disableUICapabilitiesFactory(
       };
     }, {});
 
-  const shouldDisableFeatureUICapability = (
-    featureId: keyof UICapabilities,
-    uiCapability: string
-  ) => {
-    // if the navLink isn't for a feature that we have registered, we don't wish to
-    // disable it based on privileges
-    return featureId !== 'navLinks' || featureNavLinkIds.includes(uiCapability);
+  function isCatalogueItemReferencedByFeatureSet(
+    catalogueEntry: string,
+    featureSet: Array<Partial<{ catalogue: RecursiveReadonlyArray<string> | undefined }>>
+  ) {
+    return featureSet.some((feature) => (feature.catalogue ?? []).includes(catalogueEntry));
+  }
+
+  const shouldAffectCapability = (featureId: keyof UICapabilities, uiCapability: string) => {
+    // This method answers: 'Should we affect a capability based on privileges?'
+
+    // 'spaces' and 'fileUpload' feature ID's are handled independently
+    // The spaces and file_upload plugins have their own capabilites switchers
+
+    // Always affect global settings
+    if (featureId === 'globalSettings') return true;
```

Comment:
nit: Mixing inline returns (above) with block returns (below) - I'd go with one to improve readability